tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +91,494 @@ mvn package
+`@PutMapping("/{id}")`
+
+The PUT mapping is triggered when a client sends an HTTP PUT request to `/api/v1/airline/{id}`, where `{id}` is the unique identifier of the airline to be updated.
+
+1. The client includes the updated data of the airline in the request body as a JSON object.
+2. The `AirlineController` receives the request, extracts the `id` from the URL path, and retrieves the updated airline data from the request body.
+3. The `AirlineController` calls the `updateAirline` method of the `AirlineService`, passing the `id` and updated airline data. This function internally calls [airlineCol.replace()](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html#replace) to replace the airline in the database.
+4. The `AirlineService` is responsible for updating the airline in the database using the `AirlineRepository`.
+5. If the airline is updated successfully, the `AirlineService` returns the updated airline.
+6. The `AirlineController` constructs an HTTP response with a status code of 200 OK and includes the updated airline data in the response body as a JSON object.
+7. The response is sent back to the client with the updated airline data if found, or a 404 Not Found response if the airline with the specified ID does not exist.
+
+### DELETE Mapping Workflow
+
+`@DeleteMapping("/{id}")`
+
+The DELETE mapping is triggered when a client sends an HTTP DELETE request to `/api/v1/airline/{id}`, where `{id}` is the unique identifier of the airline to be deleted.
+
+1. The `AirlineController` receives the request and extracts the `id` from the URL path.
+2. The `AirlineController` calls the `deleteAirline` method of the `AirlineService`, passing the `id` of the airline to be deleted. This function internally calls [airlineCol.remove()](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html#removing) to remove the airline from the database.
+3. The `AirlineService` is responsible for deleting the airline from the database using the `AirlineRepository`.
+4. If the airline is deleted successfully, the `AirlineService` performs the deletion operation without returning any response data.
+5. The `AirlineController` constructs an HTTP response with a status code of 204 No Content, indicating that the request was successful, but there is no content to return in the response body.
+6. The response is sent back to the client to indicate the successful deletion of the airline.
+
+These workflows illustrate how each HTTP method interacts with the `AirlineService` and the underlying database through the `AirlineRepository` to perform various operations on airline data.
+
+## Custom SQL++ Queries
+
+1. Get all airlines by country
 
 ```java
-profileCol.upsert(id, profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
+
+    @Override
+    public List<Airline> findByCountry(String country, int limit, int offset) {
```

Comment:
Also I am not sure if this is the right way to inject LIMIT & OFFSET into the query. It is prone to SQL injection if you just inject the user input into a query. You should use named parameters instead like you do for Country.

Reply:
Will have to fix this first in the other PR. Can then make the changes in this one.

Reply:
Have created a PR for the same once merged, it will have the updated changes.